core/views.py

```diff
@@ -22,105 +22,141 @@
-    
-    facility_filtered_result = metric_df[metric_df['Facility Name'].str.contains(place_result['name'],case=False)]
-    facility_filtered_result = facility_filtered_result[facility_filtered_result['relative mean'].notna()]
-    if not facility_filtered_result.empty: 
-        facility_filtered_result = facility_filtered_result.iloc[0]
-        if metric_name == "hai":
-            #this is continuous value 
-            place_result[f'{metric_name} relative mean'] = round(facility_filtered_result['relative mean'],1)
-        else:
-            #this is star value as int
-            place_result[f'{metric_name} relative mean'] = int(facility_filtered_result['relative mean'])
+@timeit
+def add_metrics_to_place(metric_df, place_result):
+    # Try to match by name
+    name_match = find_name_match(metric_df, place_result['name'])
+    if not name_match.empty:
+        for col in name_match.index:
+            place_result[col] = name_match[col]
     else:
-        #now try to match by address
-        cms_partial_address_match = find_address_match(metric_df, place_result["formatted_address"])
-        
-        if not cms_partial_address_match.empty:
-            if metric_name == "hai":
-                place_result[f'{metric_name} relative mean'] = round(cms_partial_address_match['relative mean'],1)
-            else:
-                place_result[f'{metric_name} relative mean'] = int(cms_partial_address_match['relative mean'])
-        else:
-            place_result[f'{metric_name} relative mean'] = ""
+        # Try to match by address
+        address_match = find_address_match(metric_df, place_result["formatted_address"])
+        if not address_match.empty:
+            for col in address_match.index:
+                place_result[col] = address_match[col]
     return place_result
 
 
-def find_address_match(cms_metric_df, place_address): 
-    place_address = place_address.lower()
-    place_address = place_address.replace("-", " ")
-    place_address = place_address.replace("/", " ")
-    place_address = expand_address_abbreviations(place_address)
-    for index, row in cms_metric_df.iterrows():
-        if row["Address"] in place_address: 
-            return row
+@timeit
+def find_name_match(df, name):
+    name = name.lower().strip()
+    res = df[df['Facility Name'].str.contains(name,case=False)]
+    if not res.empty:
+        match_ = res.iloc[0]
+        print(f"name match: '{name}' with '{match_['Facility Name']}'")
+        return match_
+    else:
+        # Try closest match
+        names = dict(zip(df['Facility Name'].values,df['Facility Name'].index))
+        matches = get_close_matches(name, names.keys(), n=1, cutoff=0.9)
+        if matches:
+            match_  = df.loc[names[matches[0]],:]
+            print(f"name match: '{name}' with '{match_['Facility Name']}'")
+            return match_
+    return pd.DataFrame()
+
+@timeit
+def find_address_match(cms_metric_df, place_address):
```

Comment:
Yea all this refactoring would have been best in another PR. I'm still looking for the core changes for the original ticket and its hard to find them.